### PR TITLE
Bg_Kin2_Shelf

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 1]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+s32 func_80B6FE08(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    if (this->unk_164 & (0x8 | 0x2)) {
+        return func_80B6FB30(this, globalCtx);
+    }
+    return func_80B6FCA4(this, globalCtx);
+}
+
+s32 func_80B6FE48(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    s32 params = BGKIN2SHELF_GET_1(&this->dyna.actor);
+    s32 temp_v1 = (D_80B70778[params] & this->unk_164) & 0xFF;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:152]: Maybe temp_v1 is a u8?

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 2]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    if (this->unk_164 & 1) {
+        return this->unk_167 < 1;
+    }
+
+    if (this->unk_164 & 2) {
+        return this->unk_166 < 1;
+    }
+
+    if (this->unk_164 & 4) {
+        return (this->unk_167 < 0) ^ 1;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:179]: This doesn't match with an `>=` or a `!(this->unk_167 < 0)`?

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 3]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    }
+
+    if (this->unk_164 & 8) {
+        return (this->unk_166 < 0) ^ 1;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:183]: Same

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 4]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+                                          this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+    Matrix_MultiplyVector3fByState(&sp90, &sp78);
+    Matrix_MultiplyVector3fByState(&sp84, &sp6C);
+
+    return BgCheck_EntityLineTest3(&globalCtx->colCtx, &sp78, &sp6C, &sp60, &sp50, true, false, false, true, &sp4C,
+                                   &this->dyna.actor, 0.0f);
+}
+
+s32 func_80B6FE08(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    if (this->unk_164 & (0x8 | 0x2)) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:144]:
```suggestion
    if (this->unk_164 & ((1 << 3) | (1 << 1))) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 5]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    Vec3f sp90;
+    Vec3f sp84;
+    Vec3f sp78;
+    Vec3f sp6C;
+    Vec3f sp60;
+    Vec3f sp54;
+    CollisionPoly* sp50;
+    s32 sp4C;
+
+    if (this->unk_164 & 1) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:114]:
```suggestion
    if (this->unk_164 & (1 << 0)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 6]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    Vec3f sp94;
+    Vec3f sp88;
+    Vec3f sp7C;
+    Vec3f sp70;
+    Vec3f sp64;
+    Vec3f sp58;
+    CollisionPoly* sp54;
+    s32 sp50;
+
+    if (this->unk_164 & 2) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:72]:
```suggestion
    if (this->unk_164 & (1 << 1)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 7]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    if ((this->unk_166 != 0) && (this->unk_167 != 0)) {
+        return false;
+    }
+
+    if ((this->unk_166 == 0) && (this->unk_167 == 0)) {
+        return temp_v1 != 0;
+    }
+
+    if (this->unk_166 != 0) {
+        return temp_v1 & (0x8 | 0x2);
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:163]:
```suggestion
        return temp_v1 & ((1 << 3) | (1 << 1));
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 8]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    }
+    return func_80B6FCA4(this, globalCtx);
+}
+
+s32 func_80B6FE48(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    s32 params = BGKIN2SHELF_GET_1(&this->dyna.actor);
+    u8 temp_v1 = D_80B70778[params] & this->unk_164;
+
+    if ((this->unk_166 != 0) && (this->unk_167 != 0)) {
+        return false;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:155]:
```suggestion
        return 0;
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 9]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+
+    if ((this->unk_166 == 0) && (this->unk_167 == 0)) {
+        return temp_v1 != 0;
+    }
+
+    if (this->unk_166 != 0) {
+        return temp_v1 & (0x8 | 0x2);
+    }
+
+    return temp_v1 & 5;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:166]:
```suggestion
    return temp_v1 & ((1 << 2) | (1 << 0));
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 10]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+}
+
+s32 func_80B6FEBC(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    if (this->unk_164 & 1) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:170]:
```suggestion
    if (this->unk_164 & (1 << 0)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 11]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+        if ((phi_v0_2 == this->unk_164) || (this->unk_164 == 0)) {
+            this->unk_164 = phi_v0_2;
+            this->unk_165++;
+        } else {
+            this->unk_165 = 0;
+            this->unk_164 = 0;
+        }
+
+        if (func_80B6FF28(this, globalCtx)) {
+            if (this->unk_164 & (0x8 | 0x2)) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:256]:
```suggestion
            if (this->unk_164 & ((1 << 3) | (1 << 1))) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 12]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    if (this->unk_160 >= 1.0f) {
+        Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B6FB30.s")
+        this->unk_160 = 1.0f;
+        player->stateFlags2 &= ~0x10;
+        this->dyna.pushForce = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B6FCA4.s")
+        if (this->unk_164 & 8) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:299]:
```suggestion
        if (this->unk_164 & (1 << 3)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 13]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+        this->dyna.actor.world.pos.x = ((Math_SinS(sp36) * phi_f20) * D_80B70750[sp40]) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = ((Math_CosS(sp36) * phi_f20) * D_80B70750[sp40]) + this->dyna.actor.home.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B6FEBC.s")
+        if (this->unk_166 != 0) {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+        func_80B700A8(this);
+    } else {
+        if (this->unk_164 & 8) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:315]:
```suggestion
        if (this->unk_164 & (1 << 3)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 14]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+    if (this->unk_160 >= 1.0f) {
+        Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B700C0.s")
+        this->unk_160 = 1.0f;
+        player->stateFlags2 &= ~0x10;
+        this->dyna.pushForce = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B70214.s")
+        if (this->unk_164 & 4) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:346]:
```suggestion
        if (this->unk_164 & (1 << 2)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 15]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+        this->dyna.actor.world.pos.x = ((Math_SinS(sp36) * temp_f20) * D_80B70758[sp40]) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = ((Math_CosS(sp36) * temp_f20) * D_80B70758[sp40]) + this->dyna.actor.home.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Kin2_Shelf/func_80B70498.s")
+        if (this->unk_167 != 0) {
+            play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+        func_80B700A8(this);
+    } else {
+        if (this->unk_164 & 4) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:361]:
```suggestion
        if (this->unk_164 & (1 << 2)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 16]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+
+    return temp_v1 & 5;
+}
+
+s32 func_80B6FEBC(BgKin2Shelf* this, GlobalContext* globalCtx) {
+    if (this->unk_164 & 1) {
+        return this->unk_167 < 1;
+    }
+
+    if (this->unk_164 & 2) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:174]:
```suggestion
    if (this->unk_164 & (1 << 1)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 17]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+        return this->unk_166 < 1;
+    }
+
+    if (this->unk_164 & 4) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:178]:
```suggestion
    if (this->unk_164 & (1 << 2)) {
```

File: src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c [hunk 18]
```diff
@@ -28,43 +35,345 @@ const ActorInit Bg_Kin2_Shelf_InitVars = {
...
+        return this->unk_167 >= 0;
+    }
+
+    if (this->unk_164 & 8) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Kin2_Shelf/z_bg_kin2_shelf.c:182]:
```suggestion
    if (this->unk_164 & (1 << 3)) {
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**